File: flutter_application_1/lib/services/activity_service.dart
import '../models/activity.dart';
import 'api_client.dart';

class ActivityService {
  static Future<List<Activity>> getByUserId(int userId) async {
    final json = await ApiClient.get('/activities/user/$userId');
    return (json as List? ?? [])
        .map((e) => Activity.fromJson(e as Map<String, dynamic>))
        .toList();
  }

  static Future<List<Activity>> getUserHistory(int userId) async {
    final json = await ApiClient.get('/activities/user/$userId/history');
    return (json as List? ?? [])
        .map((e) => Activity.fromJson(e as Map<String, dynamic>))
        .toList();
  }
}

File: ecolink-front/src/app/core/services/activity.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Activity } from '../models/index';
import { environment } from '../../../environments/environment';

@Injectable({ providedIn: 'root' })
export class ActivityService {
  private base = `${environment.apiUrl}/activities`;
  constructor(private http: HttpClient) {}

  create(activity: Partial<Activity>): Observable<Activity> {
    return this.http.post<Activity>(this.base, activity);
  }
  getById(id: number): Observable<Activity> {
    return this.http.get<Activity>(`${this.base}/${id}`);
  }
  getByUserId(userId: number): Observable<Activity[]> {
    return this.http.get<Activity[]>(`${this.base}/user/${userId}`);
  }
  getUserHistory(userId: number): Observable<Activity[]> {
    return this.http.get<Activity[]>(`${this.base}/user/${userId}/history`);
  }
}
